src/nbl/video/CVKLogicalDevice.h

```diff
@@ -4,20 +4,42 @@
 #include <algorithm>
 
 #include "nbl/video/ILogicalDevice.h"
+// Todo(achal): I should probably consider putting some defintions in CVKLogicalDevice.cpp
+#include "nbl/video/CVulkanCommon.h"
 #include "nbl/video/CVulkanDeviceFunctionTable.h"
 #include "nbl/video/CVKSwapchain.h"
 #include "nbl/video/CVulkanQueue.h"
+#include "nbl/video/CVulkanRenderpass.h"
+#include "nbl/video/CVulkanImageView.h"
+#include "nbl/video/CVulkanFramebuffer.h"
+#include "nbl/video/CVulkanSemaphore.h"
+#include "nbl/video/CVulkanFence.h"
+#include "nbl/video/CVulkanShader.h"
+#include "nbl/video/CVulkanSpecializedShader.h"
+#include "nbl/video/CVulkanCommandPool.h"
+#include "nbl/video/CVulkanCommandBuffer.h"
+#include "nbl/video/CVulkanDescriptorSetLayout.h"
+#include "nbl/video/CVulkanSampler.h"
+#include "nbl/video/CVulkanPipelineLayout.h"
+#include "nbl/video/CVulkanPipelineCache.h"
+#include "nbl/video/CVulkanComputePipeline.h"
+#include "nbl/video/CVulkanDescriptorPool.h"
+#include "nbl/video/CVulkanDescriptorSet.h"
+#include "nbl/video/CVulkanMemoryAllocation.h"
+#include "nbl/video/CVulkanBuffer.h"
+#include "nbl/video/CVulkanBufferView.h"
+#include "nbl/video/CVulkanForeignImage.h"
+#include "nbl/video/surface/CSurfaceVulkan.h"
 
 namespace nbl::video
 {
 
 class CVKLogicalDevice final : public ILogicalDevice
 {
 public:
-    CVKLogicalDevice(VkDevice vkdev, const SCreationParams& params, core::smart_refctd_ptr<system::ISystem>&& sys, core::smart_refctd_ptr<asset::IGLSLCompiler>&& glslc) :
-    ILogicalDevice(EAT_VULKAN, params, std::move(sys), std::move(glslc)),
-    m_vkdev(vkdev),
-    m_devf(vkdev)
+    CVKLogicalDevice(IPhysicalDevice* physicalDevice, VkDevice vkdev,
+        const SCreationParams& params, core::smart_refctd_ptr<system::ISystem>&& sys)
+        : ILogicalDevice(physicalDevice, params), m_vkdev(vkdev), m_devf(vkdev)
```

Comment:
why do you have two vkdev members?

Reply:
I guess `m_devf` is the device function table, but I have to recheck.

Reply:
ah yeah right.